### ISSUE TITLE
Unreachable code in transpiled JS

### ISSUE THREAD

**Issue by joebehymer** [Microsoft/TypeScript#13635]
**TypeScript Version:**  2.1.5 / Current Playground Version

**Code**

TS Playground Link: https://goo.gl/u9Ar8w

```ts
enum Some { Enum = 1 }
class OtherClass { }

class DeclareAfterReturn extends OtherClass {

    constructor() { 
        super();
        console.log(this.map[Some.Enum]);
    }
    
    map = {
        [Some.Enum]: {
            name: 'something'
        }
    };
}


```

**Expected behavior:**

No unreachable code

**Actual behavior:**

Unreachable variable declaration on line 26 in the Playground link above (the `var _a` after then `return`).  JSHint doesn't like this.  This changed somewhere between 2.0 and 2.1.5.

**Comment by mhegazy** [comment]
The declaration `var _a` is not a statement. it is a declaration. all declarations are hoisted to the top of their scope in JS. the tool that is reporting this error to you needs to be updated to be inline with the JS semantics, I suggest filing a bug there.

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/13592
